### ISSUE TITLE
Psalm errorBaseline config not respected when given paths

### ISSUE THREAD

**Issue by gready-hub** [vimeo/psalm#10624]
Hi there. I've found this to break how the error baseline is used. Due to the removal of the trailing `/` from the config path Psalm does not locate the files from the error baseline due too this one missing character.

Either the generate baseline needs to add the prefix or we need to remove the prefix `/` when checking if the path is in the `$issues_baseline` map.

From `src/Psalm/Internal/Cli/Psalm.php`
```php
        if ($paths_to_check !== null) {
            $filtered_issue_baseline = [];
            foreach ($paths_to_check as $path_to_check) {
                // These two lines below are the problem... 
                $path_to_check = substr($path_to_check, strlen($config->base_dir));
                if (isset($issue_baseline[$path_to_check])) {
                    $filtered_issue_baseline[$path_to_check] = $issue_baseline[$path_to_check];
                }
            }
            $issue_baseline = $filtered_issue_baseline;
        }
```

This resolves the issue I have been experiencing... Either increment by one or remove prefix `/` prior.
```
// ...
$path_to_check = substr($path_to_check, strlen($config->base_dir) + 1);
// ...
```

_Originally posted by @gready-hub in https://github.com/vimeo/psalm/issues/10542#issuecomment-1920052525_

**Comment by gready-hub** [comment]
Sample of our psalm baseline and psalm config if that's needed but I think it's clear already.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<files psalm-version="5.19.0@06b71be009a6bd6d81b9811855d6629b9fe90e1b">
  <file src="admin-scripts/test-user/configure-test-users.php">
    <InvalidCast>
      <code>$currencyId</code>
      <code>$field</code>
```

```xml
<?xml version="1.0"?>

<psalm
    autoloader="public/lib/common.php"
    errorBaseline="psalm-baseline.xml"
    errorLevel="1"
    findUnusedBaselineEntry="false"
    findUnusedCode="true"
    phpVersion="8.1"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns="https://getpsalm.org/schema/config"
    xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
>
    <plugins>
        <pluginClass class="Psalm\FreelancerPsalmPlugin\Plugin"/>
        <pluginClass class="Psalm\MockeryPlugin\Plugin"/>
        <pluginClass class="CuyZ\Valinor\QA\Psalm\ValinorPsalmPlugin"/>
    </plugins>

    <projectFiles>
        <!-- Included paths... everything! -->
        <directory name="."/>

        <ignoreFiles allowMissingFiles="true">
            <!-- External -->
            <directory name="vendor"/>
```

**Comment by weirdan** [comment]
To clarify:
```php
$path_to_check = '/a/b/c.php';
$config->base_dir = '/a/b';
$baseline_path = 'c.php'; // but checked against /c.php
```
Is that the correct problem description?

**Comment by gready-hub** [comment]
Looks correct to me.

FYI I'm running a composer patch successfully with a similar fix from the issue summary with 5.21.0, works both when specifically file paths &  without.

**Comment by gready-hub** [comment]
tyvm!
